### PR TITLE
[duplicate-code] Parse functions and class methods recursively when gathering signature lines

### LINKED ISSUE #4653: Ignore class methods signatures with similarities "ignore-signatures" option enabled
### Is your feature request related to a problem? Please describe

Initial already closed issue #3619.
One more related closed question with examples #4350.

Similarities "ignore-signatures" option currently only ignores functions' signatures.
Class methods won't be ignored with the current implementation which would be very useful to avoid false-negative `duplicate-code` errors for classes.

### Describe the solution you'd like

With similarities "ignore-signatures" option enabled, class methods' signatures are ignored the same way as functions'.

### PR DESCRIPTION
**Pull request by mhumetskyi**
Ignore class methods' and nested functions' signatures the same way as functions' with similarities "ignore-signatures" option enabled.

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :sparkles: New feature |

## Description

Closes #4653

### CODE REVIEW

File: pylint/checkers/similar.py [hunk 1]
```diff
@@ -586,11 +586,23 @@ def stripped_lines(
         }
         current_line_is_import = False
     if ignore_signatures:
-        functions = [
-            n
-            for n in tree.body
-            if isinstance(n, (astroid.FunctionDef, astroid.AsyncFunctionDef))
-        ]
+
+        def _get_functions(functions, tree):
```
Comment (Pierre-Sassoulas) [inline pylint/checkers/similar.py:590]:
```suggestion
        def _get_functions(functions: List[NodeNG], tree: astroid.NodeNG) -> List[NodeNG]:
```

**Review by Pierre-Sassoulas — APPROVED** [review]
Thank you for fixing this :) ! Could you add some typing for the new function ? We're trying to introduce it little by little.

File: pylint/checkers/similar.py (hunk 1 above)
Comment (mhumetskyi) [reply pylint/checkers/similar.py]: Sure, updated.

**Comment by Pierre-Sassoulas** [comment]
Thank you again, this will be available in 2.10.0 :)

**Pull request merged**